tests/unit/flask/test_charm.py

```diff
@@ -79,3 +81,40 @@ def test_rotate_secret_key_action(harness: Harness):
     harness.charm._on_rotate_secret_key_action(action_event)
     new_secret_key = harness.get_relation_data(0, harness.charm.app)["flask_secret_key"]
     assert secret_key != new_secret_key
+
+
+def test_log_rotate(tmp_path):
+    """
+    arrange: create a log file larger than the max log file size
+    act: run the log rotate function
+    assert: the log file has rotated
+    """
+    log_file = tmp_path / "access.log"
+    log_file.touch()
+    os.truncate(log_file, logrotate.MAX_SIZE + 1)
+    assert logrotate.rotate(str(log_file.absolute()))
+    assert not log_file.exists()
+    files = list(tmp_path.iterdir())
+    assert len(files) == 1
+    archive = files[0]
+    datetime.datetime.strptime(archive.name, "access-%Y-%m-%d-%H-%M-%S.log")
```

Comment:
To make it more readable, you might want to add a short comment explaining that we do not expect a ValueError to be raised here indicating the correct format of the archive.

Reply:
That's a good idea, updated, thanks!